app/helpers/audits_helper.rb

```diff
@@ -15,8 +15,17 @@ def id_to_label(name, change, truncate = true)
           label = name.classify.gsub('Id', '').constantize.find(change).to_label
         rescue NameError
           # fallback to the value only instead of N/A that is in generic rescue below
-          return change.to_s
+          return _("Missing(ID: %s)") % change
         end
+      when /.*_ids$/
+        existing = name.sub('_ids', '').classify.constantize.where(id: change).index_by(&:id)
+        label = change.map do |id|
+          if existing.has_key?(id)
+            existing[id].to_label
```

Comment:
this means that instead of old value we display current one, e.g. yesterday the role name was Auditor and user role_ids was set to this role, today someone renamed to role to Admin, I look at yesterday audit and I see the user was assigned Admin role, correct? is this expected behavior? I think that was the original motivation for storing values as a string, so their values are kept in history

Reply:
I think this could be considered a feature not a bug :) since it's the same object, you could argue user would expect this to point to the correct object, not the old name of the object (perhaps when we rework the page UI this could be a link to the object or its audits).

Reply:
Sounds reasonable, let me test the PR tomorrow how it feels